sdk/security_keyvault/examples/backup_secret.rs

```diff
@@ -1,24 +1,19 @@
-use azure_identity::{ClientSecretCredential, TokenCredentialOptions};
+use azure_identity::DefaultAzureCredentialBuilder;
 use azure_security_keyvault::SecretClient;
-use std::env;
-use std::sync::Arc;
+use std::{env, sync::Arc};
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let client_id = env::var("CLIENT_ID").expect("Missing CLIENT_ID environment variable.");
-    let client_secret =
-        env::var("CLIENT_SECRET").expect("Missing CLIENT_SECRET environment variable.");
-    let tenant_id = env::var("TENANT_ID").expect("Missing TENANT_ID environment variable.");
     let keyvault_url =
         env::var("KEYVAULT_URL").expect("Missing KEYVAULT_URL environment variable.");
     let secret_name = env::var("SECRET_NAME").expect("Missing SECRET_NAME environment variable.");
 
-    let creds = Arc::new(ClientSecretCredential::new(
-        tenant_id,
-        client_id,
-        client_secret,
-        TokenCredentialOptions::default(),
-    ));
+    let creds = Arc::new(
+        DefaultAzureCredentialBuilder::new()
```

Comment:
It would be great if we could switch our examples to simply `DefaultAzureCredentials::new()`. I'm curious why `.exclude_managed_identity_credential()` is being used.

Reply:
When run on a non-azure VM, this takes a *long* time to fail and move onto the next type of authentication.

Reply:
It would be great to move managed identity checking to last in the default credentials setup, but that's outside the scope of this PR.